addons/stock/models/stock_quant.py

```diff
@@ -1447,6 +1447,16 @@ def _search_owner(self, operator, value):
         else:
             return [('id', '=', False)]
 
+    def write(self, vals):
+        if 'location_id' in vals:
```

Comment:
```suggestion
        if 'location_id' in vals and vals['location_id']:
```
As `location_id` isn't required on the package, you can also set it to `False`.
While it will still work properly (i.e. the destination will be defaulted to `quant.location_id`), it will create a move and move line for nothing, as it would move the product from its initial location to its initial location.